packages/session-replay-browser/src/events/events-idb-store.ts

```diff
@@ -96,80 +86,75 @@ export const createStore = async (dbName: string) => {
+
+  constructor(args: InstanceArgs) {
+    super(args);
+    this.apiKey = args.apiKey;
+    this.db = args.db;
   }
 
-  async initialize(type: EventType, sessionId?: number) {
+  static async new(
+    type: EventType,
+    args: Omit<InstanceArgs, 'db'>,
+    sessionId?: number,
+  ): Promise<SessionReplayEventsIDBStore | undefined> {
     try {
       const dbSuffix = type === 'replay' ? '' : `_${type}`;
-      const dbName = `${this.apiKey.substring(0, 10)}_amp_session_replay_events${dbSuffix}`;
-      this.db = await createStore(dbName);
-      this.timeAtLastSplit = Date.now(); // Initialize this so we have a point of comparison when events are recorded
-      await this.transitionFromKeyValStore(sessionId);
+      const dbName = `${args.apiKey.substring(0, 10)}_amp_session_replay_events${dbSuffix}`;
+      const db = await createStore(dbName);
+      const eventsIDBStore = new SessionReplayEventsIDBStore({
+        ...args,
+        db,
+      });
+      await eventsIDBStore.transitionFromKeyValStore(sessionId);
+      return eventsIDBStore;
     } catch (e) {
-      this.loggerProvider.warn(`${STORAGE_FAILURE}: ${e as string}`);
+      args.loggerProvider.warn(`${STORAGE_FAILURE}: ${e as string}`);
     }
+    return;
   }
 
-  /**
-   * Determines whether to send the events list to the backend and start a new
-   * empty events list, based on the size of the list as well as the last time sent
-   * @param nextEventString
-   * @returns boolean
-   */
-  shouldSplitEventsList = (events: Events, nextEventString: string): boolean => {
-    const sizeOfNextEvent = new Blob([nextEventString]).size;
-    const sizeOfEventsList = new Blob(events).size;
-    if (sizeOfEventsList + sizeOfNextEvent >= this.maxPersistedEventsSize) {
-      return true;
+  async getCurrentSequenceEvents(sessionId?: number) {
+    if (sessionId) {
+      const events = await this.db.get('sessionCurrentSequence', sessionId);
+      if (!events) {
+        return undefined;
+      }
+      return [events];
     }
-    if (
-      this.timeAtLastSplit !== null &&
-      this.interval &&
-      Date.now() - this.timeAtLastSplit > this.interval &&
-      events.length
-    ) {
-      this.interval = Math.min(this.maxInterval, this.interval + this.minInterval);
-      this.timeAtLastSplit = Date.now();
-      return true;
+
+    const allEvents = [];
+    for (const events of await this.db.getAll('sessionCurrentSequence')) {
+      allEvents.push(events);
     }
-    return false;
-  };
 
-  getSequencesToSend = async () => {
+    return allEvents;
+  }
+
+  getSequencesToSend = async (): Promise<SendingSequencesReturn<number>[] | undefined> => {
     try {
-      const sequencesToSend = (await this.db?.getAll<'sequencesToSend'>(
-        sequencesToSendKey,
-      )) as SendingSequencesIDBReturn[];
+      const sequences: SendingSequencesReturn<number>[] = [];
+      let cursor = await this.db.transaction('sequencesToSend').store.openCursor();
```

Comment:
Is this so that we can use a transaction here?

Reply:
Good question! I should have left a comment, but I wanted to build out a proper list here rather than doing an unsafe type assertion. Using the cursor key I can guarantee that it'll be defined.